doc/release/upcoming_changes/20722.new_feature.rst

```diff
@@ -0,0 +1,6 @@
+Dtype with different size can be assigned to arrays with contiguous last axis
+-----------------------------------------------------------------------------
+Previously, the entire array had to be C-contiguous, which led to a lot of
+unncessary copies. This change affects both the ``dtype`` descriptor assignment
+and ``ndarray.view``. The deprecated behavior of ``ndarray.view`` which would
+allow F-contiguous arrays to be viewed is no longer possible.
```

Comment:
You should create a separate `expired` release note that mentions that this behavior has changed, preferably with an example showing the change.